sharding/sharding.go

```diff
@@ -119,8 +119,11 @@ func (r *ShardingFerry) Run() {
 	cutoverStart := time.Now()
 	// The callback must ensure that all in-flight transactions are complete and
 	// there will be no more writes to the database after it returns.
-	metrics.Measure("CutoverLock", nil, 1.0, func() {
-		err = r.config.CutoverLock.Post(client)
+	err = ghostferry.WithRetries(r.config.MaxCutoverRetries, time.Duration(r.config.CutoverRetryWaitSeconds)*time.Second, r.logger, "get cutover lock", func() (err error) {
+		metrics.Measure("CutoverLock", nil, 1.0, func() {
```

Comment:
Just a question: the metric measurement should still be on the cutover lock as opposed to surround the whole `WithRetries`?

Reply:
Right, likely summing up the individual cutover lock times, so that we get the total impact, without counting the sleeps between retries.